calibrators/marker_radar_lidar_calibrator/src/transformation_estimator.cpp

```diff
@@ -0,0 +1,188 @@
+  Eigen::Isometry3d delta_transformation = Eigen::Isometry3d::Identity();
+  delta_transformation.linear() = delta_rotation;
+
+  calibrated_radar_to_lidar_transformation_ = delta_transformation * initial_radar_to_lidar_eigen_;
+}
+
+void TransformationEstimator::estimateSVDTransformation(
+  ExtrinsicReflectorBasedCalibrator::TransformationType transformation_type)
+{
+  if (transformation_type == ExtrinsicReflectorBasedCalibrator::TransformationType::svd_2d) {
+    RCLCPP_INFO(
+      rclcpp::get_logger("marker_radar_lidar_calibrator"), "Estimate 2D SVD transformation");
+  } else {
+    RCLCPP_INFO(
+      rclcpp::get_logger("marker_radar_lidar_calibrator"), "Estimate 3D SVD transformation");
+  }
+
+  pcl::registration::TransformationEstimationSVD<PointType, PointType> estimator;
+  Eigen::Matrix4f full_radar_to_radar_optimization_transformation;
+  estimator.estimateRigidTransformation(
+    *lidar_points_ocs_, *radar_points_rcs_, full_radar_to_radar_optimization_transformation);
+  Eigen::Isometry3d calibrated_radar_to_radar_optimization_transformation(
+    full_radar_to_radar_optimization_transformation.cast<double>());
+
+  if (transformation_type == ExtrinsicReflectorBasedCalibrator::TransformationType::svd_2d) {
+    // Check that is is actually a 2D transformation
+    auto calibrated_radar_to_radar_optimization_rpy = autoware::universe_utils::getRPY(
+      tf2::toMsg(calibrated_radar_to_radar_optimization_transformation).orientation);
+    double calibrated_radar_to_radar_optimization_z =
+      calibrated_radar_to_radar_optimization_transformation.translation().z();
+    double calibrated_radar_to_radar_optimization_roll =
+      calibrated_radar_to_radar_optimization_rpy.x;
+    double calibrated_radar_to_radar_optimization_pitch =
+      calibrated_radar_to_radar_optimization_rpy.y;
+
+    if (
+      calibrated_radar_to_radar_optimization_z != 0.0 ||
+      calibrated_radar_to_radar_optimization_roll != 0.0 ||
+      calibrated_radar_to_radar_optimization_pitch != 0.0) {
+      RCLCPP_ERROR(
+        rclcpp::get_logger("marker_radar_lidar_calibrator"),
+        "The estimated 2D translation was not really 2D. Continue at your own risk. z=%.3f "
+        "roll=%.3f "
+        "pitch=%.3f",
+        calibrated_radar_to_radar_optimization_z, calibrated_radar_to_radar_optimization_roll,
+        calibrated_radar_to_radar_optimization_pitch);
+    }
+
+    calibrated_radar_to_radar_optimization_transformation.translation().z() =
+      (initial_radar_to_lidar_eigen_ * radar_optimization_to_lidar_eigen_.inverse())
+        .translation()
+        .z();
+  }
+
+  calibrated_radar_to_lidar_transformation_ =
+    calibrated_radar_to_radar_optimization_transformation * radar_optimization_to_lidar_eigen_;
+}
+
+void TransformationEstimator::estimateZeroRollTransformation()
+{
+  RCLCPP_INFO(
+    rclcpp::get_logger("marker_radar_lidar_calibrator"),
+    "Estimate 3D transformation with roll is always zero");
```

Comment:
Is this a statement ?!?!?!

Reply:
Changed the logging to https://github.com/tier4/CalibrationTools/pull/180/commits/12b356233707aa31d5be2cd629d19bbcef3401cb